uint32 hipLeft
uint32 hipRight
uint32 kneeLeft
uint32 kneeRight